caluma/caluma_core/tests/test_health_checks.py

```diff
@@ -0,0 +1,542 @@
+from django.conf import settings
+import time
+from urllib3.response import HTTPResponse
+from django.test import override_settings
+import xml
+
+from caluma.caluma_form import storage_clients
+from django.db.migrations.executor import MigrationExecutor
+
+from minio import Minio
+from minio.definitions import Object as MinioStatObject
+from minio import ResponseError
+
+from django.core import management
+
+import sys
+
+from django.db.migrations import Migration
+from watchman import settings as watchman_settings
+
+
+@pytest.mark.parametrize("working_db", [True, False])
+def test_db_connection(request, client, snapshot, minio_mock_working, working_db):
+    """
+    Test /healthz/ endpoint response when database connection is
+    available or unavailable.
+    Conditionally provides database access through the 'db' fixture.
+    This affects the watchman check for databases and the db models
+    check and prevents access to the database.
+    """
+
+    # use database fixture if test in working state
+    if working_db:
+        request.getfixturevalue("db")
+        # manually migrate db to ensure the database is in the
+        # newest state when running migration tests database
+        # might be left in an inconsistent state
+        management.call_command("migrate")
+
+    # get /healthz/ response and compare to previous snapshot
+    response = client.get("/graphql/healthz")
+    snapshot.assert_match(response.json())
+
+    # assert database check response
+    if working_db:
+        assert response.json()["databases"] == [{"default": {"ok": True}}]
+        assert response.json()["db models"] == [{"default": {"ok": True}}]
+        assert response.json()["db migrations"] == [{"default": {"ok": True}}]
+        assert response.status_code == 200
+    else:
+        assert response.json()["databases"] == [{"default": {"ok": False}}]
+        assert response.json()["db models"] == [{"default": {"ok": False}}]
+        assert response.json()["db migrations"] == [{"default": {"ok": False}}]
+        assert response.status_code == watchman_settings.WATCHMAN_ERROR_CODE
+
+
+def switch_user(user, pwd):
+    """
+    Switch database user to either 'tester' or 'caluma' and
+    re-connect to database
+    """
+
+    # check if valid user
+    users = {"tester", "caluma"}
+    if not user in users:  # pragma: no cover
+        raise Exception("Could not switch user: Invalid user name " + user)
+
+    # connect new user to database
+    connections["default"].close()
+
+    settings.DATABASES["default"]["USER"] = user
+    settings.DATABASES["default"]["PASSWORD"] = pwd
+
+    connections["default"].connect()
+
+
+def switch_user_tester():
+    """
+    Switch database user to 'tester' with password 'test'
+    """
+
+    switch_user("tester", "test")
+
+
+def switch_user_caluma():
+    """
+    Switch database user to 'tester' with password 'test'
+    """
+
+    switch_user("caluma", "caluma")
+
+
+def create_test_user():
```

Comment:
You're using this together with `remove_test_user()` in a fixture only. You could simplify this code by turning it into a fixture as well